src/overlays/actors/ovl_En_Tanron3/z_en_tanron3.c

```diff
@@ -28,39 +36,419 @@ const ActorInit En_Tanron3_InitVars = {
+        yDistance = this->targetPosWithDeviation.y - this->actor.world.pos.y;
+        zDistance = this->targetPosWithDeviation.z - this->actor.world.pos.z;
+
+        // Rotate the fish to look towards its target
+        xzDistance = sqrtf(SQ(xDistance) + SQ(zDistance));
+        atanTemp = Math_FAtan2F(xzDistance, -yDistance);
+        Math_ApproachS(&this->actor.world.rot.x, atanTemp, this->rotationScale, this->rotationStep);
+        atanTemp = Math_FAtan2F(zDistance, xDistance);
+        Math_SmoothStepToS(&this->actor.world.rot.y, atanTemp, this->rotationScale, this->rotationStep, 0);
+        Math_ApproachS(&this->rotationStep, this->targetRotationStep, 1, 0x100);
+
+        Math_ApproachF(&this->actor.speedXZ, this->targetSpeedXZ, 1.0f, this->speedMaxStep);
+        Actor_SetVelocityAndMoveXYRotationReverse(&this->actor);
+    } else {
+        switch (this->isBeached) {
+            case false:
+                // Fish is above water but hasn't touched land before
+                this->actor.gravity = -1.0f;
+                this->targetPosWithDeviation.y = this->waterSurfaceYPos - 50.0f;
+                this->workTimer[TANRON3_WORK_TIMER_OUT_OF_WATER] = 25;
+                Math_ApproachS(&this->actor.world.rot.x, 0x3000, 5, 0xBD0);
+                if (this->actor.bgCheckFlags & 8) {
+                    this->actor.speedXZ = 0.0f;
+                    if (this->actor.velocity.y > 0.0f) {
+                        this->actor.velocity.y = -1.0f;
+                    }
+                }
+                if (this->actor.bgCheckFlags & 1) {
+                    // Fish has touched land
+                    this->isBeached = true;
+                }
+                break;
+            case true:
+                this->nextRotationAngle = 0x3A98;
+                this->actor.gravity = -1.5f;
+                if (this->actor.bgCheckFlags & 1) {
+                    // Fish is still touching land, so it's still beached. Randomly flop around
+                    this->actor.velocity.y = Rand_ZeroFloat(5.0f) + 5.0f;
+                    this->actor.speedXZ = Rand_ZeroFloat(2.0f) + 2.0f;
+                    if (Rand_ZeroOne() < 0.5f) {
+                        this->targetShapeRotation.x =
+                            (s16)randPlusMinusPoint5Scaled(500.0f) + this->targetShapeRotation.x + 0x8000;
+                    }
+                    if (Rand_ZeroOne() < 0.5f) {
+                        this->targetShapeRotation.z =
+                            (s16)randPlusMinusPoint5Scaled(500.0f) + this->targetShapeRotation.z + 0x8000;
+                    }
+                    if (Rand_ZeroOne() < 0.5f) {
+                        this->targetShapeRotation.y = (s16)Rand_ZeroFloat(0x10000);
+                    }
+                    this->actor.world.rot.y = Math_FAtan2F(this->actor.world.pos.z, this->actor.world.pos.x) +
+                                              (s16)randPlusMinusPoint5Scaled(52768.0f);
```

Comment:
This is roughly 290 degrees, although not quite.

Reply:
Did you mean I should change it to something like this?
```suggestion
                                              (s16)randPlusMinusPoint5Scaled(0xCE20);
```

Reply:
It is an angle, so I think that would make sense.